controllers/imageupdateautomation_controller.go

```diff
@@ -316,15 +316,24 @@ func (r *ImageUpdateAutomationReconciler) Reconcile(ctx context.Context, req ctr
 	}
 
 	if rev, err := commitChangedManifests(tracelog, repo, tmp, signingEntity, author, message); err != nil {
-		if err == errNoChanges {
-			debuglog.Info("no changes made in working directory; no commit")
-			statusMessage = "no updates made"
-			if lastCommit, lastTime := auto.Status.LastPushCommit, auto.Status.LastPushTime; lastCommit != "" {
-				statusMessage = fmt.Sprintf("%s; last commit %s at %s", statusMessage, lastCommit[:7], lastTime.Format(time.RFC3339))
-			}
-		} else {
+		if err != errNoChanges {
 			return failWithError(err)
 		}
+
+		debuglog.Info("no changes made in working directory; no commit")
+		statusMessage = "no updates made"
+
+		if auto.Status.LastPushTime != nil && auto.Status.LastPushCommit != "" {
+			lpcLen := len(auto.Status.LastPushCommit)
+			message := `%s; last commit "%s" at %s`
+
+			if lpcLen > 7 {
+				lpcLen = 7
+				message = `%s; last commit "%s..." at %s`
+			}
+
+			statusMessage = fmt.Sprintf(message, statusMessage, auto.Status.LastPushCommit[:lpcLen], auto.Status.LastPushTime.Format(time.RFC3339))
+		}
```

Comment:
nit: could be simplified for clarity

```suggestion
		if auto.Status.LastPushTime != nil && len(auto.Status.LastPushCommit) >= 7 {
			statusMessage = fmt.Sprintf("%s; last commit %s at %s", statusMessage, auto.Status.LastPushCommit[:7], auto.Status.LastPushTime.Format(time.RFC3339))
		}
```

Reply:
I think you've missed a bit of what the new code is doing. The internal `if` on the length of the commit message is only trimming it if it's `>7`, the suggested code will only log the message if the commit message is `>7`. I can add a comment around this given that it's not clear what's happening, if that would help.

Reply:
That was intentional, as 7 characters is the short version of the hash used to represent commits. Decreasing the length here will increase the likelihood of collisions. The idea on the suggestion was to also maintain the current behaviour.

As this field represents the SHA1 commits, it should contain either the full 40 characters or 7 characters for the short version. I would assume something went wrong earlier in the process for a non-empty string with length of anything less than 7.

GitHub requires commit hashes to be at least 4 characters long (in some of their GET APIs). But I am not convinced we benefit by decreasing this.

Reply:
Ah, okay my mistake. I thought this was the commit message not the commit hash. If this is the hash then I'll make that change :+1: